src/core/lwt.mli

```diff
@@ -1400,8 +1400,23 @@ end
 (** {3 Let syntax} *)
 module Syntax :
 sig
+  (**)
```

Comment:
Would a single blank line suffice here, instead of an empty comment?

Reply:
Yep, I think it would. I default to the empty comments now because I'm used to OCamlformat killing those blank lines...

Reply:
Ah, good to know. Is that a current bug in OCamlFormat?

In any case, Lwt doesn't use it, at least not yet.

Reply:
Whether or not it's a bug is somewhat up to interpretation. OCamlformat doesn't always preserve Odoc's view of the world (in instances where Odoc is using more than just the AST structure to determine placement). It only guarantees to preserve the AST of the input program (disregarding location information).

A similar case happens for "grouped" items:

```diff
 val foo : unit
+
 val bar : unit
 (** conventional OCamlformat won't let me group these! *)
```

(although it might be possible to find some combination of options that _will_ preserve that information)

Reply:
Ok, I am taking this further off topic, but in neither case (and in no case at all, today) is odoc interpreting whitespace.

In the case of the first comment, it is the compiler's parser which associates the comment with either the module or a module item, and attaches the documentation attribute to different AST nodes according to the decision. If OCamlFormat is effectively moving the comment so that it affects that decision, it is indeed not preserving the AST, regardless of the involvement of odoc.

In the second case, that was a former interpretation of odoc, but odoc wasn't reading the whitespace or the locations. It was just interpreting sequences of items of the same kind, in which only the last had a documentation attribute, as a group. This is not analogous to the first case &mdash; it would be analogous if OCamlFormat changed the spacing in such a way, that the comment would no longer be associated with `bar` *by the compiler*.

Reply:
My understanding is that the following module types have the same AST:
```ocaml
(** Outside *)
module type S = sig
  (** Inside *)
end

(** Outside *)
module type S = sig

  (** Inside *)
end
```
In both cases, the `ocaml.text` attribute is attached as a signature item:
```
[{pstr_desc =
   Pstr_modtype
    {pmtd_name = {txt = "S"};
     pmtd_type =
      Some
       {pmty_desc =
         Pmty_signature
          [{psig_desc =
             Psig_attribute
              {attr_name = {txt = "ocaml.text"};
               attr_payload =
                PStr
                 [{pstr_desc =
                    Pstr_eval
                     ({pexp_desc =
                        Pexp_constant (Pconst_string (" Foo ", None));
                       pexp_loc_stack = []; pexp_attributes = []},
                     [])}]}}];
        pmty_attributes = []};
     pmtd_attributes = []}}]
```

Indeed, [OCamlformat would crash](https://github.com/ocaml-ppx/ocamlformat/blob/master/lib/Translation_unit.ml#L319-L334) if this were the case, since it checks for AST equality using `Mirage_ast.Printast`.

I only vaguely remember the empty comment changed the Odoc rendered output (something to do with including a module type?), but if what you say is true: perhaps neither an empty doc comment nor an empty line makes any observable difference :slightly_smiling_face: